aten/src/ATen/native/quantized/cpu/qconcat.cpp

```diff
@@ -0,0 +1,50 @@
+#include <ATen/ATen.h>
+#include <ATen/core/Type.h>
+#include <ATen/core/op_registration/op_registration.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/cpu/Loops.h>
+#include <ATen/quantized/Quantizer.h>
+
+#include <algorithm>
+#include <vector>
+
+namespace at { namespace native {
+namespace {
+template <bool ReLUFused = false>
+class QCat final : public c10::OperatorKernel {
+ public:
+  Tensor operator()(const std::vector<Tensor>& qxs, long axis,
+                    double scale, long zero_point) {
+    const auto x_dtype = qxs[0].scalar_type();
```

Comment:
do we need to verify qschemes(after qscheme diff landed) and scalar_type() all match for inputs?

Reply:
In the current implementation I don't think we need that -- we dequantize + concatenate + quantize. That way there is no need to check if they all match.

Reply:
since you are using x_dtype for output I think you need to check that.